### PR TITLE
PIPE-214 Allow FileID in context.read_file

### PR DESCRIPTION
**Pull request by yandryianets**
Also includes https://tetrascience.atlassian.net/browse/PIPE-87  (Modify context.search_eql to returns the file pointers)

Tested via self-service (by embedding ts-sdk).

---
It will be documented right after new release

### CODE REVIEW

File: ts_sdk/task/__task_script_runner.py [hunk 1]
```diff
@@ -126,6 +126,19 @@ def read_file(self, file: File, form: str = 'body') -> ReadResult:
         ...     df = pd.read_sql_query('SELECT * FROM foo', con)
         """
 
+        if len(file.keys()) == 1 and not (file['fileId'] is None):
```
Comment (chunfengd) [inline ts_sdk/task/__task_script_runner.py:129]: `file['fileId'] is not None` might be better
or `'fileId' in file`

File: ts_sdk/task/__task_script_runner.py [hunk 2]
```diff
@@ -360,9 +373,14 @@ def add_attributes(
...
         return new_file
 
+    
+
     @wrap_log('context.search_eql')
-    def search_eql(self, payload):
-        return es_datalake_search_eql(payload)
+    def search_eql(self, payload, returns='raw'):
+        search_res = es_datalake_search_eql(payload)
+        if returns == 'filePointers':
```
Comment (chunfengd) [inline ts_sdk/task/__task_script_runner.py:381]: Should we document this?

File: ts_sdk/task/__task_script_runner.py (hunk 1 above)
Comment (chunfengd) [reply ts_sdk/task/__task_script_runner.py:129]: should we document the usage of `'fileId'`?

**Review by chunfengd — COMMENTED** [review]
Can you discuss with @ChantelDean about document update for the API?

File: ts_sdk/task/__task_script_runner.py (hunk 1 above)
Comment (yandryianets) [reply ts_sdk/task/__task_script_runner.py]: sure. just after release

File: ts_sdk/task/__task_script_runner.py (hunk 2 above)
Comment (yandryianets) [reply ts_sdk/task/__task_script_runner.py:381]: sure. just after release

**Review by chunfengd — APPROVED** [review]

**Pull request merged**